### DISCUSSION TITLE
Supports different sizes of disks

### DISCUSSION THREAD

**Discussion by shivabohemian** [minio/minio#13983] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
when I start minio server with 4 drives(1t, 2t, 2t, 2t), I can't upload files if the 1t drive is almost full.

**Describe the solution you'd like**
Actually, I should still upload successfully in the other three 2t disks. Cause I'm able to recovery datas  when I change the 1t disk to 2t disk.

**Describe alternatives you've considered**

**Additional context**

**Comment by klauspost** [comment, marked as answer]
MinIO is designed around equal shard sharing and deterministic placement. 

It is not just a matter of "supporting" different sized disks; it is how MinIO is designed.

**Reply by shivabohemian** [reply]
Ok, it's the design MinIO uses, maybe better design is around  actual scene. Thanks for your answer.
